QEfficient/utils/_utils.py

```diff
@@ -119,13 +119,14 @@ def onnx_exists(model_name: str) -> Tuple[bool, str, str]:
     return onnx_exists_bool, onnx_dir_path, onnx_model_path
 
 
-def load_hf_tokenizer(model_name: str, cache_dir: Optional[str] = None, hf_token: Optional[str] = None, padding_side:str = "left", **kwargs) -> Union[PreTrainedTokenizerFast, PreTrainedTokenizer]:
+def load_hf_tokenizer(model_name: str, cache_dir: Optional[str] = None, hf_token: Optional[str] = None, local_model_dir: Optional[str] = None, padding_side:str = "left", **kwargs) -> Union[PreTrainedTokenizerFast, PreTrainedTokenizer]:
     logger.info(f"Loading Tokenizer for {model_name}")
     if hf_token is not None:
         login(hf_token)
-
+    if local_model_dir and cache_dir:
+        logger.warning(f"Both local model dir ({local_model_dir}) and cache dir ({cache_dir}) given. Using local model dir.")
     # Download tokenizer along with model if it doesn't exist
-    model_hf_path = hf_download(repo_id=model_name, cache_dir=cache_dir, allow_patterns=["*.json", "*.py", "*token*"])
+    model_hf_path = local_model_dir if local_model_dir else hf_download(repo_id=model_name, cache_dir=cache_dir, allow_patterns=["*.json", "*.py", "*token*"])
```

Comment:
Hey @quic-shagun , can we handle this without introducing this new param?
rename `model_name` parameter to `pretrained_model_name_or_path` and if it's a dir, we directly jump to loading, if it's not a dir, we download?

Reply:
We are logging info here 
```
logger.info(f"Loading Tokenizer for {model_name}")
```
If we make `model_name` as `pretrained_model_name_or_path` then the above info will not make sense for path

Reply:
Discussed offline and changed logging message